x/staking/keeper/delegation.go

```diff
@@ -614,7 +614,7 @@ func (k Keeper) Delegate(
 		}
 	}
 
-	validator, newShares = k.AddValidatorTokensAndShares(ctx, validator, bondAmt)
```

Comment:
I wonder why the validator value is not set to store?

Reply:
cc: @cwgoes @alexanderbez

Reply:
`AddValidatorTokensAndShares` stores the validator already via a call to `k.SetValidator`. This line can be `_, newShares = k.AddValidatorTokensAndShares(ctx, validator, bondAmt)`.